glommio/src/sys/uring.rs

```diff
@@ -1568,14 +1569,19 @@ impl Reactor {
         let mut main_ring = self.main_ring.borrow_mut();
         let mut lat_ring = self.latency_ring.borrow_mut();
 
+        // we need to make sure we consume the latency ring before setting the
+        // preemption timer because consuming this ring may wake up some previously
+        // asleep task queues.
+        consume_rings!(into &mut woke; lat_ring);
```

Comment:
commenting here because github doesn't allow to comment on the patch description:

`consume_rings` doesn't (usually) issue a syscall. syscalls are needed to submit new events, but the path to fetch results is syscall-free